### PR TITLE
[API Management] Revert Cross Referencing for Doc Team

### PR DESCRIPTION
**Pull request by solankisamir**
This is just adding a complete spec to what was merged earlier https://github.com/Azure/azure-rest-api-specs/pull/1049. The Docs team don't support cross-referencing yet. It is just meant for them.

Added file apimanagement-all.json which is an aggregation of all contracts

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [ ] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

### CODE REVIEW

**Comment by vishrutshah** [comment]
@solankisamir Please fix the model validator error

```
  Azure swagger model validation using x-ms-examples and examples in spec
error: Found errors in validating the response with statusCode "200" for x-ms-example "ApiManagementServiceGetNetworkStatus" in operation "NetworkStatus_GetByService".:
 
{ code: 'RESPONSE_VALIDATION_ERROR',
  message: 'Found errors in validating the response with statusCode "200" for x-ms-example "ApiManagementServiceGetNetworkStatus" in operation "NetworkStatus_GetByService".',
  innerErrors: 
   [ { code: 'INVALID_RESPONSE_BODY',
       errors: 
        [ { code: 'OBJECT_MISSING_REQUIRED_PROPERTY',
            params: [ 'error' ],
            message: 'Missing required property: error',
            path: [ 'connectivityStatus', '1' ],
            description: 'Details about connectivity to a resource.' },
          { code: 'OBJECT_MISSING_REQUIRED_PROPERTY',
            params: [ 'error' ],
            message: 'Missing required property: error',
            path: [ 'connectivityStatus', '0' ],
            description: 'Details about connectivity to a resource.' } ],
       message: 'Invalid body: Value failed JSON Schema validation',
       path: [] } ] }
{ AssertionError: swagger "arm-apimanagement/2016-10-10/swagger/apimanagement-all.json" contains model validation errors.
    at /home/travis/build/Azure/azure-rest-api-specs/test/model.js:21:14
  name: 'AssertionError',
  actual: false,
  expected: true,
  operator: '==',
  message: 'swagger "arm-apimanagement/2016-10-10/swagger/apimanagement-all.json" contains model validation errors.',
  generatedMessage: false }

```

**Review by vishrutshah — CHANGES_REQUESTED** [review]
please fix the model validation error.

**Comment by solankisamir** [comment]
@vishrutshah updated the PR.

**Comment by vishrutshah** [comment]
@solankisamir Thanks for updating it. 

I've a question for you. Can you please let us know the purpose of this swagger i.e Is it only for documentation or are you also planning to have SDKs out of this?

**Comment by solankisamir** [comment]
@vishrutshah currently this swagger is only for documentation purposes, but we want to start generating SDKs out of this. Problem currently is the OpenAPI Spec 2.0 currently doesn't fully represent our API interface. We are waiting for Open Api Spec 3.0 to be incorporated into AutoRest for complete SDK generation.

**Comment by vishrutshah** [comment]
@solankisamir Could you please fix following:

[apimdeployment.json](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-apimanagement/2016-10-10/swagger/apimdeployment.json#L74) file references `./apimanagement.json#/parameters/ResourceGroupNameParameter` but you still have that parameter defined in the same file [here](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-apimanagement/2016-10-10/swagger/apimdeployment.json#L1330)

**Comment by solankisamir** [comment]
@vishrutshah apimdeployment.json is another file, which we need not to have any cross-references. You can check here [2016-07-07](https://github.com/solankisamir/azure-rest-api-specs/tree/master/arm-apimanagement/2016-07-07/swagger)

**Comment by vishrutshah** [comment]
@solankisamir Cool I see that you are removing that referencing into this PR changes. that's great. 👍

File: arm-apimanagement/2016-10-10/swagger/apimanagement-all.json [hunk 1]
```diff
@@ -0,0 +1,7890 @@
...
+          "url": "https://msdn.microsoft.com/en-us/library/azure/dn783483.aspx"
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServiceNameParameter"
+          },
+
```
Comment (vishrutshah) [inline arm-apimanagement/2016-10-10/swagger/apimanagement-all.json]: `id`, `name` & `type` properties must be refactored into `Resource` model.

Reference: https://github.com/Azure/azure-rest-api-specs-pr/pull/40/files#diff-d3a0dcdc5d2939703fa9e32dff2295c1R3

**Review by vishrutshah — CHANGES_REQUESTED** [review]
There were RPCViolations detected from the linter. 

```
BodyTopLevelProperties | M3006 | ApiContract | Extra Properties : "ApiContract/serviceUrl, ApiContract/path, ApiContract/protocols, ApiUpdateContract/serviceUrl, ApiUpdateContract/path, ApiUpdateContract/protocols"
BodyTopLevelProperties | M3006 | OperationContract| Extra Properties: “OperationContract/method, OperationContract/urlTemplate, OperationUpdateContract/method, OperationUpdateContract/urlTemplate”
OperationsAPIImplementationValidation | M3023| Operations API must be implemented for '/providers/Microsoft.ApiManagement/operations’
ProvidersPathValidation | Violating Path - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/tenant/configuration/syncState
ProvidersPathValidation | Violating Path - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/tenant/configuration/validate
. . .
```

@solankisamir Did you receive any approved exceptions on these?

File: arm-apimanagement/2016-10-10/swagger/apimanagement-all.json [hunk 2]
```diff
@@ -0,0 +1,7890 @@
...
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The response body contains the specified API entity.",
+            "schema": {
+              "$ref": "#/definitions/ApiContract"
+            },
+            "headers": {
```
Comment (solankisamir) [inline arm-apimanagement/2016-10-10/swagger/apimanagement-all.json]: @vishrutshah  The ETag response-header field provides the current value of the entity tag for the requested variant. The headers used with entity tags are described in sections [14.24](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.24), [14.26](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.26) and [14.44](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.44)

File: arm-apimanagement/2016-10-10/swagger/apimanagement-all.json [hunk 3]
```diff
@@ -0,0 +1,7890 @@
...
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/apis/{apiId}": {
+      "get": {
+        "tags": [
+          "Apis"
+        ],
+        "operationId": "Apis_Get",
+        "description": "Gets the details of the API specified by its identifier.",
+        "produces": [
+          "application/json",
```
Comment (solankisamir) [inline arm-apimanagement/2016-10-10/swagger/apimanagement-all.json]: @vishrutshah yes the api returns these formats, when making a request with [Accept](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.19) header specified for these format.

File: arm-apimanagement/2016-10-10/swagger/apimanagement-all.json (hunk 1 above)
Comment (ravbhatnagar) [inline arm-apimanagement/2016-10-10/swagger/apimanagement-all.json]: This violates Resource Provider Contract (RPC). @solankisamir - Were your APIs reviewed and approved by the ARM team/Azure API review board.

Comment (solankisamir) [inline arm-apimanagement/2016-10-10/swagger/apimanagement-all.json]: @ravbhatnagar we will update our contracts for the next version of the API. Can you provide us exception for this API version?

**Comment by ravbhatnagar** [comment]
@vishrutshah - had an email conversation with @solankisamir and they will address the RPC violations in the next API version update. You can ignore those for now.

**Comment by vishrutshah** [comment]
Issue https://github.com/Azure/azure-rest-api-specs/issues/1060 for reference

**Review by vishrutshah — APPROVED** [review]

**Comment by solankisamir** [comment]
#sign-off

**Pull request merged**